src/bhattacharyya.jl

```diff
@@ -50,6 +50,34 @@ end
     return sqab, asum, bsum
 end
 
+@inline function _bhattacharyya_coeff(a::SparseVectorUnion{<:Number}, b::SparseVectorUnion{<:Number})
+    anzind = nonzeroinds(a)
+    bnzind = nonzeroinds(b)
+    anzval = nonzeros(a)
+    bnzval = nonzeros(b)
+    ma = nnz(a)
+    mb = nnz(b)
+
+    ia = 1; ib = 1
+    eval_op = (a, b) -> sqrt(a * b)
+    s = zero(eval_op(zero(eltype(a)), zero(eltype(b))))
```

Comment:
I overlooked the other occurence of `eval_op`, but maybe spell it out and make it consistent with the dense version above?
```suggestion
    s = zero(typeof(sqrt(oneunit(eltype(a))*oneunit(eltype(b)))))
```

Reply:
I couldn't decide how to set the type of `s`. I suppose if the methods for `sqrt` and `*` for the given input types are type stable, then it  doesn't matter. But, maybe there is a reason to prefer 1 over 0.

Reply:
IIRC, then that had to do with units. Like you want the right output unit, including the squaring and taking the square root. It could be, though, that this unit-logic works just as fine with `zero`, I don't recall.